adev/src/app/editor/download-manager.service.ts

```diff
@@ -50,4 +51,20 @@ export class DownloadManager {
     anchor.click();
     anchor.remove();
   }
+
+  private async generateZip(files: FileAndContent[]): Promise<Uint8Array> {
+    const filesObj: Record<string, Uint8Array> = {};
+    files.forEach(({path, content}) => {
+      filesObj[path] = typeof content === 'string' ? strToU8(content) : content;
+    });
+
+    console.log(filesObj);
+    const content = new Promise<Uint8Array>((resolve) => {
+      zip(filesObj, (err, data) => {
+        resolve(data);
+      });
+    });
+
+    return await content;
```

Comment:
Stale `console.log` and any error is propagated; you'd at least need something like but I'm not sure what the error experience would be in that case

```suggestion
    return new Promise((resolve, reject) => {
      zip(filesObj, (err, data) => {
        if (err) {
          reject(err);
        } else {
          resolve(data);
        }
      });
    });
```

Reply:
The medium-term goal is the bring back the docs package into the framework repo. 
So I went with this instead. 

Also we didn't have any error handling ATM, so I went without it.